server/pkg/webserver/service/importer/taskdb.go

```
@@ -0,0 +1,56 @@
+package importer
+
+import (
+	"fmt"
+
+	"github.com/vesoft-inc/nebula-studio/server/pkg/config"
+
+	"go.uber.org/zap"
+	"gorm.io/driver/sqlite"
+	"gorm.io/gorm"
+	"gorm.io/gorm/logger"
+)
+
+type TaskDb struct {
+	*gorm.DB
+}
+
+/*
+	`InitDB` initialize local sql by open sql and create task_infos table
+*/
+func InitDB() {
+	dbFilePath := config.Cfg.Web.SqlitedbFilePath
+	//os.Remove(dbFilePath)
+	db, err := gorm.Open(sqlite.Open(dbFilePath), &gorm.Config{
+		Logger: logger.Default.LogMode(logger.Info)})
+	if err != nil {
+		zap.L().Fatal(fmt.Sprintf("init db fail: %s", err))
+	}
+	err = db.AutoMigrate(&TaskInfo{})
+	if err != nil {
+		zap.L().Fatal(fmt.Sprintf("init taskInfo table fail: %s", err))
+	}
+	GetTaskMgr().db = &TaskDb{
+		DB: db,
+	}
+}
+
+func (t *TaskDb) InsertTaskInfo(info *TaskInfo) {
+	t.Create(info)
+}
+
+func (t *TaskDb) UpdateTaskInfo(info *TaskInfo) {
+	t.Model(&TaskInfo{}).Where("id = ?", info.ID).Updates(info)
+}
+
+func (t *TaskDb) DelTaskInfo(ID int) {
+	t.Delete(&TaskInfo{}, ID)
+}
+
+func (t *TaskDb) LastId() int {
+	var id int
+	if t.Raw("SELECT MAX(id) FROM task_infos").Scan(&id).Error != nil {
+		return 0
+	}
+	return id
+}
```

Is it need to return an error? Using `1`(`0 + 1`) as the `id` when `error != nil`?

---

If there is no data in the table, an error will occur here. The self incrementing ID will not be 0.

---

So why you return `0`?